GeneralsMD/Code/Libraries/Source/WWVegas/WW3D2/dx8wrapper.h

```diff
@@ -1130,6 +1135,12 @@ WWINLINE void DX8Wrapper::Clamp_Color(Vector4& color)
 		cmovnb edi,edx
 		mov dword ptr[esi+12],edi
 	}
+#else
+	for (int i=0;i<4;++i) {
+		float f=(color[i]<0.0f) ? 0.0f : color[i];
+		color[i]=(f>1.0f) ? 1.0f : f;
+	}
+#endif // defined(_MSC_VER) && _MSC_VER < 1300
 }
```

Comment:
Maybe this could be reshuffled to:
```
#if defined(_MSC_VER) && _MSC_VER < 1300
	if (CPUDetectClass::Has_CMOV_Instruction()) {
		// asm code
		return;
	}
#endif // defined(_MSC_VER) && _MSC_VER < 1300
// c++ code
```
to avoid code duplication (and #else preprocessor block)

Reply:
Yes good idea. Fixed.